motionSetPointsApp/src/tests/testSetPoint.cc

```diff
@@ -0,0 +1,451 @@
+
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 2);
+
+        std::vector<double> coords{ 42.1, 77.9 };
+        int positionFound = posn2name(coords, 0.2, envFilename, posDiff);
+
+        ASSERT_EQ(positionFound, 0);
+        ASSERT_NEAR(posDiff, pow(pow(0.1, 2) + pow(0.1, 2), 0.5), 0.0000001);
+
+        getPosnName(positionName, false, envFilename);
+        ASSERT_STREQ(positionName, "Position_42_78");
+    }
+
+    TEST(setPoint, GIVEN_two_coordinate_position_that_does_not_match_existing_position_within_tolerance_WHEN_posn2name_called_THEN_current_table_row_pointer_set_correctly) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position_42_78    42   78", "Position_39_63    39   63" };
+        MockFileIO mockFile;
+        double posDiff;
+        char positionName[200];
+
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 2);
+
+        std::vector<double> coords{ 40.0, 77 };
+        int positionFound = posn2name(coords, 0.1, envFilename, posDiff);
+
+        ASSERT_EQ(positionFound, -1);
+
+        getPosnName(positionName, false, envFilename);
+        ASSERT_STREQ(positionName, "");
+    }
+
+    TEST(setPoint, GIVEN_three_coordinate_position_that_matches_existing_position_within_tolerance_WHEN_posn2name_called_THEN_current_table_row_pointer_set_correctly) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position_42_78_98    42   78  98", "Position_39_63_15    39   63   15" };
+        MockFileIO mockFile;
+        double posDiff;
+        char positionName[200];
+        std::vector<double> position {39.1, 63.1, 14.9};
+
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 3);
+
+        int positionFound = posn2name(position, 0.2, envFilename, posDiff);
+
+        ASSERT_EQ(positionFound, 0);
+        ASSERT_NEAR(posDiff, pow(3 * pow(0.1, 2), 0.5), 0.0000001);
+
+        getPosnName(positionName, false, envFilename);
+        ASSERT_STREQ(positionName, "Position_39_63_15");
+    }
+
+    TEST(setPoint, GIVEN_name_for_three_coordinate_position_WHEN_name2posn_called_THEN_RBV_table_row_pointer_set_correctly) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position_42_78_98    42   78  98", "Position_39_63_15    39   63   15" };
+        MockFileIO mockFile;
+        double position;
+
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 3);
+
+        int positionFound = name2posn("Position_42_78_98", envFilename);
+
+        ASSERT_EQ(positionFound, 0);
+
+        getPosn(0, true, envFilename, position);
```

Comment:
Suggestion: remove variable reassignment and have instead position1, position2 etc.